pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PrecisPublishLimiter.java

```diff
@@ -29,17 +30,20 @@
     protected volatile long publishMaxByteRate = 0;
     protected volatile boolean publishThrottlingEnabled = false;
     // precise mode for publish rate limiter
-    private RateLimiter topicPublishRateLimiterOnMessage;
-    private RateLimiter topicPublishRateLimiterOnByte;
+    private volatile HashMap<String, RateLimiter> rateLimiters;
```

Comment:
What is the reason for using a HashMap to store the RateLimiter instances? 

This is a bit problematic solution from thread safety perspective. `volatile` on a java.util.HashMap field won't help to ensure thread safety.

Reply:
The reason for the HashMap is to allow the possibility of new RateLimiters.

Reply:
I don't think there is justification for this HashMap solution. It simply doesn't provide any value currently. We don't have a use case for new ratelimiters at the moment. The solution can be added for that use case when it is needed.

Reply:
Reverted in the latest commit.